src/http/multipart.rs

```diff
@@ -14,14 +14,19 @@ impl CreateAttachment {
     }
 }
 
+#[derive(Clone, Debug)]
+pub enum MultipartUpload {
+    /// A file sent with the form data as an individual upload. For example, a sticker.
+    File(CreateAttachment),
+    /// Files sent with the form as message attachments.
+    Attachments(Vec<CreateAttachment>),
+}
+
 /// Holder for multipart body. Contains files, multipart fields, and payload_json for creating
 /// requests with attachments.
 #[derive(Clone, Debug)]
 pub struct Multipart {
-    /// Files that are sent with the form data as individual uploads.
-    pub upload_file: Option<CreateAttachment>,
-    /// Files that are sent with the form data as message attachments.
-    pub attachment_files: Option<Vec<CreateAttachment>>,
+    pub upload: MultipartUpload,
```

Comment:
Should this still be called `files`? If not, we might want to adjust the docstring on `Multipart` somehow.